spec/requests/memberships_controller_spec.rb

```diff
@@ -318,5 +318,18 @@
         expect(response).to have_http_status(:forbidden)
       end
     end
+
+    context 'current_user is the group leader' do
+      include_context 'current_user is the group leader'
+      context 'when membership has DEFAULT post' do
+        let(:membership) { create(:membership, :newbie) }
+        it 'rejects the membership' do
+          post "/groups/#{membership.group.id}/memberships/#{membership.id}/withdraw", xhr: true
+
+          expect(response).to have_http_status(:ok)
+          expect(Membership).not_to exist(id: membership.id)
+        end
+      end
+    end
```

Comment:
I'm not sure this exact test case is necessary. Happy path is already tested and I don't see any added value by this meanwhile request specs are expensive. I'd recommend to extend the policy spec to check the action is permitted for the group leader as well.